gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -643,8 +643,12 @@ var gpii = fluid.registerNamespace("gpii");
         var userToken = finalPayload.userToken,
             lifecycleInstructions = finalPayload.activeConfiguration.lifecycleInstructions;
         var session = that.getSession([userToken]);
-        if (!session) {
-            fluid.fail("User with token " + userToken + " has no active session");
+        if (!session) { // if user has logged out since the update was added to queue
+            var msg = "User with token " + userToken + " has no active session, so igoring update request";
+            fluid.log(msg);
+            var promise = fluid.promise();
```

Comment:
This can just be return fluid.promise().reject(msg); although in general we despise chaining

Reply:
Ah, that's better...